src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/PoolingDataSource.cs

```diff
@@ -33,6 +35,42 @@ internal sealed class PoolingDataSource : SqlDataSource
         private static int _objectTypeCount; // EventSource counter
         private readonly int _objectID = Interlocked.Increment(ref _objectTypeCount);
 
+        //TODO: readonly TimeSpan _connectionLifetime;
+
+        // Counts the total number of open connectors tracked by the pool.
+        volatile int _numConnectors;
```

Comment:
_numConnectors is modifiable by multiple threads. My understanding is that we need this to prevent compiler optimizations that may affect correctness (for example, using a stale value). https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile

Reply:
Hmm, I see, so it mostly works the same as it would in the realm I'm familiar with it being used in. My follow up question is whether this is better, worse, or the same as accessing it via Interlocked methods?

But also - we should still add `private` to it.

Reply:
Done!